app/Nova/RemoteAttendanceLink.php

```diff
@@ -0,0 +1,209 @@
+                }),
+
+            Text::make('Auto-redirecting Link', 'secret')
+                ->onlyOnDetail()
+                ->resolveUsing(static function (string $secret): string {
+                    return route('attendance.remote.redirect', ['secret' => $secret]);
+                })
+                ->canSee(static function (Request $request): bool {
+                    if (isset($request->resourceId)) {
+                        $resource = RAL::find($request->resourceId);
+                        if (null !== $resource && is_a($resource, RAL::class)) {
+                            return null !== $resource->redirect_url;
+                        }
+                    }
+
+                    return false;
+                })
+                ->readonly(static function (Request $request): bool {
+                    return true;
+                }),
+
+            Text::make('Non-redirecting Link', 'secret')
+                ->onlyOnDetail()
+                ->resolveUsing(static function (string $secret): string {
+                    return route('attendance.remote', ['secret' => $secret]);
+                })
+                ->readonly(static function (Request $request): bool {
+                    return true;
+                }),
+
+            Text::make('Secret')
+                ->onlyOnForms()
+                ->default(hash('sha256', random_bytes(64)))
+                ->canSee(static function (Request $request): bool {
+                    return $request->user()->can('update-remote-attendance-links');
+                })
+                ->creationRules('unique:remote_attendance_links,secret')
+                ->updateRules('unique:remote_attendance_links,secret,{{resourceId}}')
+                ->help('This is contained in the attendance URL that will be shared. The default value for this field'.
+                    ' is randomly generated.'),
+
+            DateTime::make('Expires At')
+                ->default(Carbon::now()->addHours(4))
+                ->help('This defaults to four hours in the future.'),
+
+            Text::make('Redirect URL')
+                ->sortable()
+                ->required(false)
+                ->rules('nullable', 'url')
+                ->help('If you put a link to a BlueJeans or Google Meet meeting here, everyone who clicks the '.
+                    'attendance link will be redirected to that meeting after their attendance is recorded. If '.
+                    'you add a redirect URL, do not share that URL directly. Only Google Meet and '.
+                    'BlueJeans calls are supported currently in the user-facing action.'),
+
+            Text::make('Note')
+                ->hideFromIndex()
+                ->required(false)
+                ->help('This can be used to keep track of what this link was used for more specifically. Press the '.
+                    'down arrow for suggestions.')
+                ->suggestions(self::$recommendedNotes),
```

Comment:
You don't want to use this on the action?

Reply:
No; in fact, I'll switch this to use the select-or-other component as well. I want to strongly encourage using one of the suggestions.